### PR TITLE
[togglesg] New extractor for toggle.sg

### PR DESCRIPTION
**Pull request by ping**
Toggle.sg is an online service from a broadcast station in Singapore.

The extractor works for clips and programmes available for streaming on the site, example http://video.toggle.sg/en/series/lion-moms-tif/trailers/lion-moms-premier/343115

Some content are geo-restricted or drm-protected.

### CODE REVIEW

File: youtube_dl/extractor/togglesg.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    _API_USER = 'tvpapi_147'
+    _API_PASS = '11111'
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id, note='Downloading video page')
+
+        api_user = self._search_regex(
+            r'apiUser:\s*"([^"]+)"', webpage, 'apiUser', default=self._API_USER, fatal=False)
```
Comment (dstftw) [inline youtube_dl/extractor/togglesg.py]: There is no need in `fatal` when `default` is provided.

File: youtube_dl/extractor/togglesg.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    determine_ext,
+    parse_iso8601,
+    remove_end
+)
+from ..compat import compat_urllib_request
+
+
+class ToggleSgIE(InfoExtractor):
+    IE_NAME = 'togglesg'
+    _VALID_URL = r'https?://video\.toggle\.sg/(?:(en|zh))/(?:(series|clips|movies))/.+?/(?P<id>[0-9]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/togglesg.py]: Don't capture groups if you are not going to use them.

**Comment by ping** [comment]
Changes made.

**Comment by dstftw** [comment]
Cherry-picked, thanks. Sorry for long wait.

**Pull request closed without merging**